### DISCUSSION TITLE
Configuring Consent Flow

### DISCUSSION THREAD

**Discussion by gamemaker1** [Sunbird-RC/community#136] [Issues] [answered]
Hi,

I am trying to use the consent flow of the registry, but have not been able to do that so far. I created an example registry - creating/inviting, retrieving, updating, making claims and attesting claims works perfectly. 

I added `school` as an optional scope for the `registry-frontend` client and created a `User Property` mapper for it. I also enabled the `Consent Required` option for the `registry-frontend` client in keycloak's admin console.

As part of the consent flow:
- I created an entity in keycloak
- Opened [this](https://localhost:8080/auth/realms/sunbird-rc/protocol/openid-connect/auth?client_id=registry-frontend&redirect_uri=*&response_type=code&scope=openid%20school) URL:

```
https://localhost:8080/auth/
  realms/sunbird-rc/
  protocol/openid-connect/auth/
    ?client_id=registry-frontend
    &redirect_uri=*
    &response_type=code
    &scope=openid school
```

- I logged in as the entity (username and default password) and expected to see a consent screen. However, I was immediately redirected to the client application with a code in the URL query parameters.

Could someone please guide me as to how I can enable the consent screen when requesting access to an entity's data? Thank you.

// @dileepbapat @kesavanp123 @tejash-jl

**Comment by tejash-jl** [comment, marked as answer]
Can you check if consent is enabled for that client @gamemaker1 ?
You would see an option called **Consent Required**

**Reply by gamemaker1** [reply]
Yes, I have enabled it in the `realm-export.json` file [here](https://github.com/gamemaker1/sunbird-rc-core/blob/main/tools/cli/src/templates/config/keycloak/realm-export.json#L551) and [here](https://github.com/gamemaker1/sunbird-rc-core/blob/main/tools/cli/src/templates/config/keycloak/realm-export.json#L579).

**Reply by gamemaker1** [reply]
Thanks @tejash-jl. I just found the issue - I had enabled `consentRequired` for the `admin-api` client, not the `registry-frontend` client. Once I enabled it for `registry-frontend` as well, I could see the consent screen when the client application requested it.

Thank you.
